status-reporter/status_reporter/status_reporter.py

```diff
@@ -138,18 +135,21 @@ def _report(self):
         # If there is a malfunctioning manager,
         # let's try to avoid using the same manager always.
         random.shuffle(self._managers_ips)
+        reporting_result = False
         for manager_ip in self._managers_ips:
             try:
-                self._report_status(manager_ip, report)
-                self._update_managers_ips_list(manager_ip)
-                return
+                client = self._get_cloudify_http_client(manager_ip)
+                reporting_result = self._report_status(client, report)
             except Exception as e:
                 logger.debug('Error had occurred while trying to report '
-                             'status and update active managers-ips list: {}'
+                             'status: {0}'
                              .format(e))
-        logger.error('Could not find an active manager to '
-                     'report the current status,'
-                     ' tried %s', ','.join(self._managers_ips))
+        if reporting_result:
+            self._update_managers_ips_list(client)
```

Comment:
Maybe move it after `self._report_status(client, report)` and then the `reporting_result` boolean it redundant.

Reply:
Not really, it will be the same like before